service/dps/storm/common/src/main/java/eu/europeana/cloud/service/dps/storm/NotificationBolt.java

```diff
@@ -124,14 +124,25 @@ private void notifyTask(NotificationTuple notificationTuple, NotificationCache n
         int processesFilesCount = nCache.getProcessed();
         int errors = nCache.getErrors();
 
+        //notification table (for single record)
         storeNotification(processesFilesCount, taskId,
                 notificationTuple.getParameters());
 
         if (error) {
             storeNotificationError(taskId, nCache, notificationTuple);
         }
-        if ((processesFilesCount % PROCESSED_INTERVAL) == 0) {
+        if(isCounterUpdateRequired(nCache)){
+            LOGGER.info("Updating task counter for task_id = {} and counter value: {}", taskId, processesFilesCount);
             taskInfoDAO.setUpdateProcessedFiles(taskId, processesFilesCount, errors);
+            nCache.setLastCounterUpdate(new Date());
+        }
```

Comment:
Shouldn't this be synchronized somehow to prevent multiple threads from concluding simultaneously that a counter update is required? Or are we accepting this risk?

Reply:
There is only one bolt instance (and only one thread) that will update the counter, so there shouldn't be any concurrency issues, we don't need to synchronize it.